src/data/reconstructing_datatypes.jl

```diff
@@ -425,39 +425,86 @@ constructrr(::JLDFile, ::Type{T}, dt::CompoundDatatype, ::Vector{ReadAttribute})
 
 
 ## Type reconstruction
+abstract type AbstractReconstructedType{N} end
 
-module ReconstructedTypes end
+struct ReconstructedPrimitive{N, T} <: AbstractReconstructedType{N}
+    val::T
+end
+function Base.show(io::IO, f::ReconstructedPrimitive{N}) where {N}
+    print(io, "Reconstruct@$N($(Int(f.val)))")
+end
 
-isreconstructed(x) = isreconstructed(typeof(x))
 
-function isreconstructed(::Type{T}) where T
-    return supertype(T) == Any && parentmodule(T) == ReconstructedTypes
+struct ReconstructedSingleton{N} <: AbstractReconstructedType{N} end
+
+function Base.show(io::IO, ::ReconstructedSingleton{N}) where {N}
+    print(io, "Reconstruct@$N()")
+end
+
+struct ReconstructedStatic{N, FN, NT} <: AbstractReconstructedType{N}
+    fields::NamedTuple{FN, NT}
+end
+
+function Base.getproperty(rc::ReconstructedStatic, s::Symbol) 
+    getproperty(getfield(rc, 1),s)
+end
+
+# simple one-line display (without trailing line break)
+function Base.show(io::IO, f::ReconstructedStatic{N, FN, FT}) where {N,FN,FT}
+    print(io, "Reconstruct@$N($(values(getfield(f,1))))")
 end
 
+struct ReconstructedMutable{N, FN, FT} <: AbstractReconstructedType{N}
+    fields::Vector{Any}
+end
+
+function Base.getproperty(rc::ReconstructedMutable{N,FN,FT}, s::Symbol) where {N,FN,FT}
```

Comment:
you'll have to overwrite `propertynames` as well, else it defaults to using the field names (which are not accessible here)